File: strategy/src/main/java/com/nearsoft/workshop/strategy/character/Queen.java
package com.nearsoft.workshop.strategy.character;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Queen extends Character {
    private final Logger log = LoggerFactory.getLogger(this.getClass());

    @Override
    public void fight() {
        log.debug(String.format("Mama! just killed a man, and I'm %s", weaponBehaviour.useWeapon()));
    }
}

File: strategy/src/main/scala/com/nearsoft/workshop/strategy/scala/character/Queen.scala
package com.nearsoft.workshop.strategy.scala.character

import org.slf4j.LoggerFactory

class Queen extends Character {

  val log = LoggerFactory.getLogger(classOf[King])

  override def fight = {
    log.debug(s"Mama! just killed a man, and I'm ${useWeapon}")
  }
}
